### ISSUE TITLE
iOS - Build project Failed (Cannot load underlying module)

### ISSUE THREAD

**Issue by BenRx** [jesster2k10/react-native-giphy-ui#2]
Hello,

I'm having an issue with your library. I added it to a new project via the yarn add command but when I try to build the xcode project I get a "Cannot load underlying module for 'DeepDiff'" error.
I tried to manually link the lib but same result. :/
Any idea ?

Thanks !

Xcode version : 11.5
Swift version : 5
React-native version : 0.62.2

<img width="274" alt="Capture d’écran 2020-05-29 à 12 44 48" src="https://user-images.githubusercontent.com/19388794/83252022-d481a580-a1aa-11ea-8608-51848a395283.png">

**Comment by jesster2k10** [comment]
Hey, could you show me your Podfile? It’s likely to do with that.

**Comment by BenRx** [comment]
Thank you! It's a basic podfile generated by react-native. I didn't make any particular modifications. :/

`platform :ios, '10.0'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

def add_flipper_pods!(versions = {})
  versions['Flipper'] ||= '~> 0.33.1'
  versions['DoubleConversion'] ||= '1.1.7'
  versions['Flipper-Folly'] ||= '~> 2.1'
  versions['Flipper-Glog'] ||= '0.3.6'
  versions['Flipper-PeerTalk'] ||= '~> 0.0.4'
  versions['Flipper-RSocket'] ||= '~> 1.0'

  pod 'FlipperKit', versions['Flipper'], :configuration => 'Debug'
  pod 'FlipperKit/FlipperKitLayoutPlugin', versions['Flipper'], :configuration => 'Debug'
  pod 'FlipperKit/SKIOSNetworkPlugin', versions['Flipper'], :configuration => 'Debug'
  pod 'FlipperKit/FlipperKitUserDefaultsPlugin', versions['Flipper'], :configuration => 'Debug'
  pod 'FlipperKit/FlipperKitReactPlugin', versions['Flipper'], :configuration => 'Debug'

  # List all transitive dependencies for FlipperKit pods
  # to avoid them being linked in Release builds
  pod 'Flipper', versions['Flipper'], :configuration => 'Debug'
  pod 'Flipper-DoubleConversion', versions['DoubleConversion'], :configuration => 'Debug'
  pod 'Flipper-Folly', versions['Flipper-Folly'], :configuration => 'Debug'
  pod 'Flipper-Glog', versions['Flipper-Glog'], :configuration => 'Debug'
  pod 'Flipper-PeerTalk', versions['Flipper-PeerTalk'], :configuration => 'Debug'
  pod 'Flipper-RSocket', versions['Flipper-RSocket'], :configuration => 'Debug'
  pod 'FlipperKit/Core', versions['Flipper'], :configuration => 'Debug'
  pod 'FlipperKit/CppBridge', versions['Flipper'], :configuration => 'Debug'
  pod 'FlipperKit/FBCxxFollyDynamicConvert', versions['Flipper'], :configuration => 'Debug'
  pod 'FlipperKit/FBDefines', versions['Flipper'], :configuration => 'Debug'
  pod 'FlipperKit/FKPortForwarding', versions['Flipper'], :configuration => 'Debug'
  pod 'FlipperKit/FlipperKitHighlightOverlay', versions['Flipper'], :configuration => 'Debug'
  pod 'FlipperKit/FlipperKitLayoutTextSearchable', versions['Flipper'], :configuration => 'Debug'
  pod 'FlipperKit/FlipperKitNetworkPlugin', versions['Flipper'], :configuration => 'Debug'
end

# Post Install processing for Flipper
def flipper_post_install(installer)
  installer.pods_project.targets.each do |target|
    if target.name == 'YogaKit'
      target.build_configurations.each do |config|
        config.build_settings['SWIFT_VERSION'] = '4.1'
      end
    end
  end
end

target 'AwesomeProject' do
  # Pods for AwesomeProject
  pod 'FBLazyVector', :path => "../node_modules/react-native/Libraries/FBLazyVector"
  pod 'FBReactNativeSpec', :path => "../node_modules/react-native/Libraries/FBReactNativeSpec"
  pod 'RCTRequired', :path => "../node_modules/react-native/Libraries/RCTRequired"
  pod 'RCTTypeSafety', :path => "../node_modules/react-native/Libraries/TypeSafety"
  pod 'React', :path => '../node_modules/react-native/'
  pod 'React-Core', :path => '../node_modules/react-native/'
  pod 'React-CoreModules', :path => '../node_modules/react-native/React/CoreModules'
  pod 'React-Core/DevSupport', :path => '../node_modules/react-native/'
  pod 'React-RCTActionSheet', :path => '../node_modules/react-native/Libraries/ActionSheetIOS'
  pod 'React-RCTAnimation', :path => '../node_modules/react-native/Libraries/NativeAnimation'
  pod 'React-RCTBlob', :path => '../node_modules/react-native/Libraries/Blob'
  pod 'React-RCTImage', :path => '../node_modules/react-native/Libraries/Image'
  pod 'React-RCTLinking', :path => '../node_modules/react-native/Libraries/LinkingIOS'
  pod 'React-RCTNetwork', :path => '../node_modules/react-native/Libraries/Network'
  pod 'React-RCTSettings', :path => '../node_modules/react-native/Libraries/Settings'
  pod 'React-RCTText', :path => '../node_modules/react-native/Libraries/Text'
  pod 'React-RCTVibration', :path => '../node_modules/react-native/Libraries/Vibration'
  pod 'React-Core/RCTWebSocket', :path => '../node_modules/react-native/'

  pod 'React-cxxreact', :path => '../node_modules/react-native/ReactCommon/cxxreact'
  pod 'React-jsi', :path => '../node_modules/react-native/ReactCommon/jsi'
  pod 'React-jsiexecutor', :path => '../node_modules/react-native/ReactCommon/jsiexecutor'
  pod 'React-jsinspector', :path => '../node_modules/react-native/ReactCommon/jsinspector'
  pod 'ReactCommon/callinvoker', :path => "../node_modules/react-native/ReactCommon"
  pod 'ReactCommon/turbomodule/core', :path => "../node_modules/react-native/ReactCommon"
  pod 'Yoga', :path => '../node_modules/react-native/ReactCommon/yoga', :modular_headers => true

  pod 'DoubleConversion', :podspec => '../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec'
  pod 'glog', :podspec => '../node_modules/react-native/third-party-podspecs/glog.podspec'
  pod 'Folly', :podspec => '../node_modules/react-native/third-party-podspecs/Folly.podspec'

  target 'AwesomeProjectTests' do
    inherit! :complete
    # Pods for testing
  end

  use_native_modules!

  # Enables Flipper.
  #
  # Note that if you have use_frameworks! enabled, Flipper will not work and
  # you should disable these next few lines.
  add_flipper_pods!
  post_install do |installer|
    flipper_post_install(installer)
  end
end

target 'AwesomeProject-tvOS' do
  # Pods for AwesomeProject-tvOS

  target 'AwesomeProject-tvOSTests' do
    inherit! :search_paths
    # Pods for testing
  end
end`

**Comment by jesster2k10** [comment]
My best guess is that the issue stems from the GiphySDK library using Embedded Swift Libraries which doesn't play nice witch default cocoa pods settings.

What you can do to solve this (hopefully) is either:
- Add `use_frameworks` to the top of your pod file. This would break support for Flipper, so I'd image you wouldn't want to do this, unless you wanted to get rid of Flipper.

- The second thing you could do is add the following to the `target 'AwesomeProject' do` section of your podfile.

```ruby
pod 'Giphy', :modular_headers => true
```

And run `cd ios && rm -rf Podfile.lock && rm -rf Pods && pod install`

That should hopefully solve the issue without you needing to add `use_frameworks!` to all your pods and potentially breaking other things.

It gives the same effect as use_frameworks! but scoped to a single pod.

You might need to do the same for any of the `Giphy` pod dependencies.

**Comment by BenRx** [comment]
Thank you for your prompt response!

I tried both of your solutions (completely remove Flipper and add the ```use_frameworks!``` mention and tried to add the ```pod giphy, :modular_headers => true``` mention) and in both cases xcode found new errors with the giphy SDK. :/

<img width="267" alt="Capture d’écran 2020-05-29 à 13 23 41" src="https://user-images.githubusercontent.com/19388794/83255474-c8004b80-a1b0-11ea-95f5-83ce62f46d40.png">

**Comment by jesster2k10** [comment]
Are you able to send me a code example that I could test locally? I'm not sure why this issue is arising. When you added `use_frameworks!` what was the output ofter cleaning the build folder

**Comment by BenRx** [comment]
Yes, of course! I quickly created a github project with my sources.
You'll find it here: https://github.com/BenRx/GiphyIntegTest

**Comment by jesster2k10** [comment]
Okay so I've managed to narrow down the issue. The second method I described actually has no effect, in order for this library to work, `use_frameworks!` has to be enabled. There's no workaround at the moment.

The other compiler errors that show up when `use_frameworks` is enabled has to do with a change in the underlying native SDK. They changed around some of the code in a newer version of the pod which leads to the errors since the current implementation no longer would work.

So, with that said, what can be done to fix it is:
- Update the library to reflect the changes in the native SDK
- Install an older version of the `Giphy` pod without these changes.

I'll stick with the first and likely push some changed later today to reflect this.
